src/service/resources-impl.js

```diff
@@ -1272,6 +1313,23 @@ export class Resources {
       }
     }
   }
+
+  /**
+   * Removes the task resource from the discovered first viewport
+   * resources array.
+   * @param {!TaskDef} task
+   */
+  checkForFirstViewportElementsWhenTaskDone_(task) {
```

Comment:
Just a suggestion:

Instead of even more state vars for this already crazy class, do it like this:

- Have the `Resource` expose a promise for when they finished loading.
- Add a method to `Resources` called `getResourcesInViewport`
- Then we filter the response why `prerenderAllowed` and by state (whether layout started)
- And then we join all the load promises for those resources

That way the `Resources` class knows nothing about the thing we are measuring, and we can easily add more metrics. CC @dvoytenko